dask/dataframe/optimize.py

```diff
@@ -60,7 +66,9 @@ def optimize_read_parquet_getitem(dsk, keys):
                 # ... with a single item...
                 return dsk
 
-            if list(block.dsk.values())[0][0] != operator.getitem:
+            if list(block.dsk.values())[0][
+                0
+            ] != operator.getitem and not block.output.startswith("iloc"):
```

Comment:
I'm open to suggestions if there's a better way to check for a `read_parquet -> iloc`

Reply:
To be sure: does the current optimization work with `loc`, which is really identical to column selection if the other selector is `:`?

Reply:
I doubt that it works with `.loc`.

Reply:
> I'm open to suggestions if there's a better way to check for a read_parquet -> iloc

Probably out of scope for this PR, but I think we'll want to specialize the kind of `Block` that we return from an `iloc` operation. This feels similar to https://github.com/dask/dask/pull/6261 where I added things like `BlockwiseGetitem` to avoid having to dive into `Blockwise` objects to figure out what they are.